### PR TITLE
polygon/bridge: Remove reader duplication

### LINKED ISSUE #11673: Remove code duplication between `bridge.Bridge` and `bridge.Reader`
`bridge.Bridge` and `bridge.Reader` has two implementations of `Events` which are almost identical.

Depends on #11626

### PR DESCRIPTION
**Pull request by shohamc1**
Switches bridge to store last event IDs.

Fixes #11673

### CODE REVIEW

File: polygon/bridge/reader.go [hunk 1]
```diff
@@ -46,25 +47,22 @@ func NewReader(store Store, log log.Logger, stateReceiverContractAddress string)
 func (r *Reader) Events(ctx context.Context, blockNum uint64) ([]*types.Message, error) {
 	start, end, err := r.store.EventIDRange(ctx, blockNum)
 	if err != nil {
-		return nil, err
-	}
-
-	if end == 0 { // exception for tip processing
```
Comment (antonis19) [inline polygon/bridge/reader.go:52]: Why are you no longer checking this?

File: polygon/bridge/reader.go [hunk 2]
```diff
@@ -46,25 +47,22 @@ func NewReader(store Store, log log.Logger, stateReceiverContractAddress string)
...
 		}
+
+		return nil, err
 	}
 
 	eventsRaw := make([]*types.Message, 0, end-start+1)
 
 	// get events from DB
-	events, err := r.store.Events(ctx, start+1, end+1)
+	events, err := r.store.Events(ctx, start, end+1)
```
Comment (antonis19) [inline polygon/bridge/reader.go:60]: What's the reason for this change?

Comment (shohamc1) [reply polygon/bridge/reader.go:60]: `EventIDRange` now returns [start, end] instead of (start, end]. Events is [start, end).

File: polygon/bridge/reader.go (hunk 1 above)
Comment (shohamc1) [reply polygon/bridge/reader.go:52]: Because end can no longer be 0. `ErrEventIDRangeNotFound` is returned if the map is not found.

File: polygon/bridge/bridge.go [hunk 3]
```diff
@@ -63,10 +61,12 @@ type Bridge struct {
 	borConfig                    *borcfg.BorConfig
 	eventFetcher                 eventFetcher
 	stateReceiverContractAddress libcommon.Address
+
+	reader *Reader
```
Comment (antonis19) [inline polygon/bridge/bridge.go:65]: Could you rename this to `eventReader` and the struct to `EventReader`. I think this gives a clearer indication of  what we are reading.

Comment (shohamc1) [reply polygon/bridge/bridge.go:65]: It is not just an event reader, we can get the block number from transaction hash too. I think `EventReader` is a misnomer. 

`Reader` contains all the read-only operations for the bridge. Since it is part of the `bridge` package, it will be referenced as `bridge.Reader` outside the package, which I think is clear.

**Review by taratorio — APPROVED** [review]

File: polygon/bridge/bridge.go (hunk 3 above)
Comment (antonis19) [reply polygon/bridge/bridge.go:65]: Ah, I see. I think it is `func (r *Reader) EventTxnLookup(ctx context.Context, borTxHash libcommon.Hash) (uint64, bool, error) {` that confused me. Could you add a short description for it?
